src/rdb_protocol/pseudo_time.cc

```diff
@@ -215,16 +215,22 @@ std::string time(const std::string &s) {
                  strprintf("Time string `%s` must have 0 or 2 colons.", s.c_str()));
     mandatory_digits(s, 2, &at, &out);
     if (optional_char(s, '.', &at, &out)) {
-        size_t read = 0;
-        while (at < s.size() && read < 3) {
-            mandatory_digits(s, 1, &at, &out);
-            read += 1;
-        }
-        while (at < s.size()) {
-            mandatory_digits(s, 1, &at, NULL);
-        }
-        while (read++ < 3) {
-            out += '0';
+        if (reql_version < reql_version_t::v2_4) {
```

Comment:
Why does version matter here? Can't it just be changed to always do this behavior?

Reply:
Regular queries run with the latest reql_version value, but old secondary indexes' functions need to run with whatever reql_version value they were created with, in order to work properly.

Reply:
The upgrade process states that [secondary indexes need to be rebuilt during upgrade](https://rethinkdb.com/docs/migration/#rebuild-indexes), so I don't see why this needs to be mainted since it's part of 2.4 and will require an upgrade.

Reply:
It says that secondary indexes *should* be rebuilt.

Reply:
Just trying to keep the code simpler and avoid potential bugs down the road. Sorry for the noise.

Reply:
It's no problem.  Eventually the old pre-2.4 behavior will be dropped (being optimistic about future releases happening here...)